### ISSUE TITLE
API to override css classes in components

### ISSUE THREAD

**Issue by GeorgeTaveras1231** [xogroup/union#306]
Per concern expressed in [this comment](https://github.com/xogroup/union/issues/301#issuecomment-322895880) and #260, we need a consistent and flexible way to override styles in a component.

**Comment by marioharper** [comment]
I like the idea. I think this means we would now need to treat class names as contracts. Be consistent with class naming so that they're more predictable. Example: component wrapper class should always be called `container`, etc. 

A working POC would be nice to play with if you can throw one together quickly on a branch.

**Comment by GeorgeTaveras1231** [comment]
@marioharper @peterkle @markmachine check out the POC at #319

**Comment by hsbacot** [comment]
I agree that we should handle styles across components in a more uniform way. CSS provides a hierarchy that we should all be familiar with. Classes should be able to be overridden by the later classes applied. All `classNames` should be able to be overridden via the `style` prop for inline styles. We shouldn't deviate from this. If it is necessary to change how styles are applied to meet our component needs, we should look at our component API's.

I think these issues point towards a solution that would involve building components that support more composition. This would allow components to be expose more granular control of the styles and allow more granular control of elements in a natural way.

**Comment by marioharper** [comment]
I do like a composition based API more. But this would mean our consumers (xo engineers) would need to open PRs on Union to add specific functionality. For example having checkbox labels rendering on the left side https://github.com/xogroup/union/issues/301. This is definitely a use case that would probably come up on other projects so contributing it to union is beneficial. But would this impede their development process?
